LockExample/FileService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Text;
+namespace LockExample
+{
+    internal static class FileService
+    {
+        private const string ResultFileName = "result.txt";
+        private const string FolderName = "LockStatementApp";
+        private static readonly object fileLock = new();
+
+        public static string GetPathDirectory(string dirFilename)
+        {
+            var dirApp = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            var locationFile = Path.Combine(dirApp, FolderName, dirFilename);
+            var dir = Path.GetDirectoryName(locationFile);
+            if (!Directory.Exists(dir))
+            {
+                if (dir is not null)
+                {
+                    _ = Directory.CreateDirectory(dir);
+                }
+            }
+            return dir ?? "";
+        }
+        public static void UpdateFile(string fileName)
```

Comment:
Consider implementing the following:
- the method that takes the folder to copy as parameter and destination
- the method should copy all the files in directory
- once the copy of file is completed the result should be written to the result file: source file name, target file name, size, time when copy is completed, time it takes to copy the file. 

Consider using async methods where available.

Please also add the tests for this method.